server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -2091,6 +2092,45 @@ protected long transportDelayMillis(String actionName) {
         }
     }
 
+    public void testGetClusterFormationStateDoesNotBlockOnMutex() {
+        // Pick any node in the cluster (all have a Coordinator)
+        try (Cluster cluster = new Cluster(1)) {
+            cluster.stabilise();
+            final Coordinator coordinator = cluster.getAnyNode().coordinator;
+
+            // Latches for coordination
+            final var mutexAcquired = new CountDownLatch(1);
+            final var release = new CountDownLatch(1);
```

Comment:
Consider using a `new CyclicBarrier(2)` for this kind of release-and-wait/wait-and-release pattern, it's a little shorter.